x-pack/plugins/actions/server/saved_objects/migrations.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  SavedObjectMigrationMap,
+  SavedObjectUnsanitizedDoc,
+  SavedObjectMigrationFn,
+} from '../../../../../src/core/server';
+import { RawAction } from '../types';
+import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
+
+export function getMigrations(
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup
+): SavedObjectMigrationMap {
+  return { '7.10.0': addHasAuthConfigurationObject(encryptedSavedObjects) };
+}
+
+const addHasAuthConfigurationObject = (
+  encryptedSavedObjects: EncryptedSavedObjectsPluginSetup
+): SavedObjectMigrationFn<RawAction, RawAction> => {
+  return encryptedSavedObjects.createMigration<RawAction, RawAction>(
```

Comment:
To handle errors and prevent Kibana from crashing during the migration process, we'll need some try/catch logic here in case one of these connectors fail to decrypt or anything else that may cause an error to be thrown.